icon-validator/validate-icon.c

```diff
@@ -132,9 +132,17 @@ rerun_in_sandbox (const char *arg_width,
   g_autofree char *err = NULL;
   int status;
   g_autoptr(GError) error = NULL;
-  char validate_icon[256];
+  char validate_icon[PATH_MAX + 1];
+  ssize_t symlink_size;
 
-  readlink ("/proc/self/exe", validate_icon, 256);
+  symlink_size = readlink ("/proc/self/exe", validate_icon, sizeof (validate_icon) - 1);
```

Comment:
There's also `glnx_readlinkat_malloc()`.

Reply:
Well, there's also `flatpak_readlink()`. I was trying to avoid linking the icon validator against more than absolutely necessary. Maybe thats silly